### PR TITLE
fix: decimal handling in Read component (st-691) 

### PR DESCRIPTION
**Pull request by ericHgorski**

### CODE REVIEW

File: src/pages/stake/ValidatorActions.tsx [hunk 1]
```diff
@@ -39,16 +39,19 @@ const ValidatorActions = ({ destination }: { destination: ValAddress }) => {
   }
 
   const renderDelegationsValue = () => {
+    if (!chain) return null
     const amount =
       delegationState.isSuccess && delegation
         ? delegation.balance.amount.toString()
         : "0"
 
+    console.log(delegation && delegation.balance.amount.toString())
```
Comment (restorenode) [inline src/pages/stake/ValidatorActions.tsx]: log statement should be deleted.

Comment (ericHgorski) [reply src/pages/stake/ValidatorActions.tsx]: fixed

File: src/pages/stake/StakedDonut.tsx [hunk 2]
```diff
@@ -64,24 +64,22 @@ const StakedDonut = ({ chain }: { chain?: string }) => {
...
-              token={payload[0]?.payload.denom}
-            />
+            <Read amount={amount} fixed={2} denom={denom} />
           </p>
         </div>
         <div className={styles.infoLine}>
           <p>Value: </p>
           <p>
-            <Read amount={payload[0]?.payload.value} fixed={2} decimals={0} />{" "}
+            <Read amount={value} denom={denom} fixed={2} decimals={0} />{" "}
```
Comment (restorenode) [inline src/pages/stake/StakedDonut.tsx]: Didn't we get rid of the need to pass decimals to the `Read` component?

File: src/pages/stake/components/StakedCard.tsx [hunk 3]
```diff
@@ -37,7 +37,8 @@ const StakedCard = (props: PropsWithChildren<Props>) => {
           style={{ justifyContent: "space-between", alignItems: "flex-end" }}
         >
           <span className={styles.value}>
-            {currency.symbol} <Read amount={value} decimals={0} fixed={2} />
+            {currency.symbol}{" "}
+            <Read amount={value} denom="" decimals={0} fixed={2} />
```
Comment (restorenode) [inline src/pages/stake/components/StakedCard.tsx:41]: Decimals included.  May want to just have `fixed` default to 2 in the `Read` component

File: src/pages/wallet/Asset.tsx [hunk 4]
```diff
@@ -74,7 +74,6 @@ const Asset = (props: Props) => {
                   decimals={decimals}
                   fixed={2}
                   denom=""
-                  token=""
                 />
```
Comment (restorenode) [inline src/pages/wallet/Asset.tsx:77]: Decimals included.  Consider making `fixed` default to 2 inside of the `Read` component.

**Review by restorenode — CHANGES_REQUESTED** [review]
Maintain consistency in the usage of and the passing of attributes to the `Read` component.

File: src/pages/stake/StakedDonut.tsx (hunk 2 above)
Comment (ericHgorski) [reply src/pages/stake/StakedDonut.tsx]: this for a currency amount, not a token denom. You can still override the decimals of the `read` component by passing. I experimented with a number of ways to simplify props passing here but this is the best and most flexible.

File: src/pages/stake/components/StakedCard.tsx (hunk 3 above)
Comment (ericHgorski) [reply src/pages/stake/components/StakedCard.tsx:41]: this also for a currency amount, not a token denom.

File: src/pages/wallet/Asset.tsx (hunk 4 above)
Comment (ericHgorski) [reply src/pages/wallet/Asset.tsx:77]: fixed can both be a boolean and a number

**Review by restorenode — CHANGES_REQUESTED** [review]
Incorrect asset showing under Swap page `Balance` and `Balance after tx`.  Make sure to recheck the changes you made again on each component.  They should really each be checked while in the process of development.

Screenshot 2023-08-15 at 11 48 40 PM

**Comment by ericHgorski** [comment]
> Incorrect asset showing under Swap page `Balance` and `Balance after tx`. Make sure to recheck the changes you made again on each component. They should really each be checked while in the process of development.
> 
> Screenshot 2023-08-15 at 11 48 40 PM

fixed! thanks for your attention to detail.

**Review by restorenode — CHANGES_REQUESTED** [review]
Token symbol not supposed to be displaying after amounts on token cards in Staking rewards:
image

For reference, this is how it displays in production:
image

**Comment by ericHgorski** [comment]
> Token symbol not supposed to be displaying after amounts on token cards in Staking rewards: image
> 
> For reference, this is how it displays in production: image

fixed

**Review by restorenode — CHANGES_REQUESTED** [review]
There are tokens that have `0.00` staking rewards that are being displayed under `Staking rewards`.  This does not happen in production.

image

**Review by restorenode — CHANGES_REQUESTED** [review]
Token symbol being displayed after amount in proposals:
image

In production:
image

**Comment by ericHgorski** [comment]
> Token symbol being displayed after amount in proposals: image
> 
> In production: image

fixed both issues

**Review by restorenode — CHANGES_REQUESTED** [review]
Clicking on the first proposal in the `Voting` section, 313.8 million does not match the 31.38 billion referenced next to `Voted:`.

Screenshot 2023-08-17 at 9 32 09 PM

**Comment by ericHgorski** [comment]
> Clicking on the first proposal in the `Voting` section, 313.8 million does not match the 31.38 billion referenced next to `Voted:`.
> 
> Screenshot 2023-08-17 at 9 32 09 PM

good catch! fixed.

**Comment by restorenode** [comment]
> > Clicking on the first proposal in the `Voting` section, 313.8 million does not match the 31.38 billion referenced next to `Voted:`.
> > Screenshot 2023-08-17 at 9 32 09 PM
> 
> good catch! fixed.

I believe the value should have been 313.9M.  `Yes` commands 214.4M and this value should be 68.30% of the total.

`214.4M / 313.9M ≈ 0.6830 or 68.30%`

Screenshot 2023-08-17 at 10 50 25 PM

**Comment by restorenode** [comment]
> > > Clicking on the first proposal in the `Voting` section, 313.8 million does not match the 31.38 billion referenced next to `Voted:`.
> > > Screenshot 2023-08-17 at 9 32 09 PM
> > 
> > 
> > good catch! fixed.
> 
> I believe the value should have been 313.9M. `Yes` commands 214.4M and this value should be 68.30% of the total.
> 
> `214.4M / 313.9M ≈ 0.6830 or 68.30%`
> 
> Screenshot 2023-08-17 at 10 50 25 PM

Nevermind, it looks like the values inside the vote options are incorrect as well and need updating.  The value inside of `Yes` should be `21.44B`.

`21.44B / 31.39B ≈ 0.6830 or 68.30%`

**Comment by ericHgorski** [comment]
> > > > Clicking on the first proposal in the `Voting` section, 313.8 million does not match the 31.38 billion referenced next to `Voted:`.
> > > > Screenshot 2023-08-17 at 9 32 09 PM
> > > 
> > > 
> > > good catch! fixed.
> > 
> > 
> > I believe the value should have been 313.9M. `Yes` commands 214.4M and this value should be 68.30% of the total.
> > `214.4M / 313.9M ≈ 0.6830 or 68.30%`
> > Screenshot 2023-08-17 at 10 50 25 PM
> 
> Nevermind, it looks like the values inside the vote options are incorrect as well and need updating. The value inside of `Yes` should be `21.44B`.
> 
> `21.44B / 31.39B ≈ 0.6830 or 68.30%`

fixed

**Review by restorenode — CHANGES_REQUESTED** [review]
Token symbol shouldn't display after dollar value in staking tooltip:
Screenshot 2023-08-17 at 11 37 14 PM

In production:
Screenshot 2023-08-17 at 11 38 31 PM

**Comment by ericHgorski** [comment]
> Token symbol shouldn't display after dollar value in staking tooltip: Screenshot 2023-08-17 at 11 37 14 PM
> 
> In production: Screenshot 2023-08-17 at 11 38 31 PM

fixed

File: src/pages/wallet/AssetChain.tsx [hunk 5]
```diff
@@ -129,13 +125,7 @@ const AssetChain = (props: Props) => {
...
                   <span className="danger">{t("Failed to query balance")}</span>
                 ) : (
-                  <Read
-                    {...props}
-                    amount={balance}
-                    token=""
-                    fixed={2}
-                    decimals={decimals}
-                  />
+                  <Read {...props} amount={balance} fixed={2} />
```
Comment (restorenode) [inline src/pages/wallet/AssetChain.tsx]: Duplicate token symbol:

Screenshot 2023-08-18 at 10 17 31 AM

File: src/pages/wallet/Vesting.tsx [hunk 6]
```diff
@@ -23,15 +23,15 @@ const Vesting = () => {
         <dl>
           <dt>{t("Vested")}</dt>
           <dd>
-            <Read amount={schedule.amount.vested} />
+            <Read amount={schedule.amount.vested} denom={denom} />
           </dd>
           <dt>{t("Total")}</dt>
           <dd>
-            <Read amount={schedule.amount.total} />
+            <Read amount={schedule.amount.total} denom={denom} />
```
Comment (restorenode) [inline src/pages/wallet/Vesting.tsx]: Token symbol displaying on branch:

Screenshot 2023-08-18 at 10 21 22 AM


Production:

Screenshot 2023-08-18 at 10 21 55 AM

File: src/pages/wallet/VestingScheduleTable.tsx [hunk 7]
```diff
@@ -24,7 +28,7 @@ const VestingScheduleTable = ({ type, schedule }: ParsedVestingSchedule) => {
         {
           title: "Amount",
           dataIndex: "amount",
-          render: (value) => <Read amount={value} />,
+          render: (value) => <Read amount={value} denom={denom} />,
```
Comment (restorenode) [inline src/pages/wallet/VestingScheduleTable.tsx]: Token symbol shouldn't display in VestingScheduleTable:
Screenshot 2023-08-18 at 10 24 07 AM

Production:

Screenshot 2023-08-18 at 10 24 54 AM

File: src/txs/Tx.tsx [hunk 8]
```diff
@@ -375,7 +367,6 @@ function Tx<TxValues>(props: Props<TxValues>) {
           <dd>
             {gasFee.amount && (
               <Read
-                decimals={decimals}
                 {...gasFee}
```
Comment (restorenode) [inline src/txs/Tx.tsx:370]: Double check decimals of `Fee` vs send token.  On branch:

Screenshot 2023-08-18 at 10 29 40 AM


Production:

Screenshot 2023-08-18 at 10 30 15 AM

File: src/txs/gov/VoteForm.tsx [hunk 9]
```diff
@@ -56,6 +58,7 @@ const VoteForm = () => {
     estimationTxValues,
     createTx,
     chain,
+    token: baseAsset,
```
Comment (restorenode) [inline src/txs/gov/VoteForm.tsx:61]: Double check vote form `Fee` value.  On branch:

Screenshot 2023-08-18 at 10 37 06 AM


In production:

Screenshot 2023-08-18 at 10 37 35 AM

**Review by restorenode — CHANGES_REQUESTED** [review]

File: src/pages/wallet/VestingScheduleTable.tsx (hunk 7 above)
Comment (ericHgorski) [reply src/pages/wallet/VestingScheduleTable.tsx]: fixed

File: src/pages/wallet/Vesting.tsx (hunk 6 above)
Comment (ericHgorski) [reply src/pages/wallet/Vesting.tsx]: fixed

File: src/txs/Tx.tsx (hunk 8 above)
Comment (ericHgorski) [reply src/txs/Tx.tsx:370]: these are incorrect on prod currently and correct here

File: src/txs/gov/VoteForm.tsx (hunk 9 above)
Comment (ericHgorski) [reply src/txs/gov/VoteForm.tsx:61]: these are incorrect on prod currently and correct here

File: src/txs/swap/TFMSwapForm.tsx [hunk 10]
```diff
@@ -309,12 +306,8 @@ const TFMSwapForm = ({ chainID }: { chainID: string }) => {
...
               addonAfter={
                 <AssetReadOnly>
-                  {simulatedValue ? (
-                    <Read
-                      amount={simulatedValue}
-                      decimals={askDecimals}
-                      approx
-                    />
+                  {simulatedValue && askAsset ? (
+                    <Read denom={askAsset} amount={simulatedValue} approx />
```
Comment (restorenode) [inline src/txs/swap/TFMSwapForm.tsx]: Asset symbol shouldn't display after amount in swap form:

Screenshot 2023-08-18 at 11 44 33 AM

File: src/pages/wallet/AssetChain.tsx [hunk 11]
```diff
@@ -129,18 +125,11 @@ const AssetChain = (props: Props) => {
...
```
Comment (restorenode) [inline src/pages/wallet/AssetChain.tsx]: Allow for IBCed token to display with native symbol:

Screenshot 2023-08-18 at 11 55 46 AM

**Review by restorenode — CHANGES_REQUESTED** [review]

File: src/txs/swap/TFMSwapForm.tsx (hunk 10 above)
Comment (ericHgorski) [reply src/txs/swap/TFMSwapForm.tsx]: fixed

File: src/pages/wallet/AssetChain.tsx (hunk 11 above)
Comment (ericHgorski) [reply src/pages/wallet/AssetChain.tsx]: fixed

**Review by restorenode — APPROVED** [review]
I will approve this PR as I am unable to find further issues.  I believe a developer should also go through the changes and approve this PR as well.  This is due to the fact that this branch is changing a lot of things inside the application and there are possibilities of missed errors.

**Comment by ericHgorski** [comment]
> I will approve this PR as I am unable to find further issues.  I believe a developer should also go through the changes and approve this PR as well.  This is due to the fact that this branch is changing a lot of things inside the application and there are possibilities of missed errors.

Yea Im starting to question the approach I took here. Im less convinced its making a meaningful positive change to the codebase given the complexity its adding. A  Read and ReadToken tiered component approach would be much cleaner.

**Pull request closed without merging**